tests/services/test_scheduler.py

```
@@ -164,6 +164,57 @@ def test_sets_permissions_error_on_cold_location_when_not_writable(self, db_sess
+        """A critically full cold drive should not be mislabeled as missing write permission."""
+        from app.services.scheduler import check_storage_permissions_job_func
+
+        cold_path = tmp_path / "cold"
+        cold_path.mkdir()
+        loc = self._make_cold_location(db_session, str(cold_path))
+
+        # Simulate a full drive where write checks fail because no free blocks remain.
+        monkeypatch.setattr("app.services.scheduler.os.access", lambda path, mode: mode != 2)
+        monkeypatch.setattr(
+            "app.services.scheduler.shutil.disk_usage",
+            lambda path: (100, 96, 4),
```

Ruff (0.15.11)

[warning] Unused lambda argument: `path`

(ARG005)